test/e2e/clusterctl_upgrade.go

```diff
@@ -292,6 +292,17 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 			input.PreUpgrade(managementClusterProxy)
 		}
 
+		// Get the workloadCluster before the management cluster is upgraded to make sure that the upgrade did not trigger
+		// any unexpected rollouts.
+		preUpgradeMachineList := &clusterv1alpha3.MachineList{}
+		err = managementClusterProxy.GetClient().List(
+			ctx,
+			preUpgradeMachineList,
+			client.InNamespace(testNamespace.Name),
+			client.MatchingLabels{clusterv1.ClusterLabelName: workLoadClusterName},
```

Comment:
```suggestion
			client.MatchingLabels{clusterv1.ClusterLabelName: workloadClusterName},
```
nit

Reply:
This variable was not introduced in this PR. Will have to change in a few places to address this nit. Might make more sense to do a broader clean up in a separate PR.

Reply:
Up to you

Reply:
But it's fine we can just leave it as is, just saw it when reviewing.